### PR TITLE
Implement gRPC integration configuration

### PR DESCRIPTION
**Pull request by delner**
This pull request implements the configuration core from #450 in the gRPC integration.

### CODE REVIEW

File: spec/ddtrace/contrib/grpc/interception_context_spec.rb [hunk 1]
```diff
@@ -60,7 +60,7 @@
 
         specify do
           expect(span.name).to eq 'grpc.client'
-          expect(span.span_type).to eq 'grpc'
```
Comment (pawelchcki) [inline spec/ddtrace/contrib/grpc/interception_context_spec.rb]: why do we want to change the `span_type` here? My main/only concern if it won't negatively impact the users of this integration. And maybe it seems a bit out of scope for this PR>

Comment (delner) [reply spec/ddtrace/contrib/grpc/interception_context_spec.rb]: It was something I worked with the author over before, and I think I made a mistake in saying that it should be `grpc`. gRPC is a framework, not a protocol. It'd be like saying Faraday spans are span_type `faraday`, which makes the tag kind of useless when it only reflects the name of the integration it came from. `http` is most fitting here because gRPC uses the HTTP/2 protocol to conduct its business anyways.

Comment (pawelchcki) [reply spec/ddtrace/contrib/grpc/interception_context_spec.rb]: GRPC is its more like that builds upon http/2 than a client like Faraday, just like http builds on TCP. 
Still I think if this change is made - it should be done explicitly using separate PR.

Comment (delner) [reply spec/ddtrace/contrib/grpc/interception_context_spec.rb]: This span type is only consumed on the Datadog side, and by moving it here, we're making it more consistent with all the other integrations that are being upgraded by doing it in lockstep. Regardless, such a new PR would funnel into the same `upgrade_to_integration_configuration` PR anyways, and be released at the same time. I would advise keeping this in.

Comment (pawelchcki) [reply spec/ddtrace/contrib/grpc/interception_context_spec.rb]: Ok sounds, good.
